keps/sig-network/4559-kubelet-probes/README.md

```diff
@@ -0,0 +1,1037 @@
+  network probes]. Apparently this slowness is a somewhat inevitable
+  side effect of how exec probes work in an OCI runtime context?
+  Possibly because it requires making multiple synchronous calls?
+  Anyway, it's possible this could be improved.
+
+- **Network Probe Performance (?)**: In the past, there have been problems
+  with probes consuming unexpectedly large amounts of kubelet resources
+  ([kubernetes #89898], "Sometime Liveness/Readiness Probes fail because
+  of net/http: request canceled while waiting for connection
+  (Client.Timeout exceeded while awaiting headers)"). This has been
+  mitigated to some extent ([kubernetes #115143])... it's not clear if
+  this is still considered a problem.
+
+See further discussion in [kubernetes #102613].
+
+#### RESOLVED Summary discussions (preserving for history)
+- Figure out if there are any concerns with non-namespace-based Linux
+  runtimes (kata, kubevirt, etc): There are no concerns here because both kata containers and [kubevirt] use network namespaces. They expect the CNI to set up the pod networking. Then their agent creates the required networking infrastructure inside this pod network namespace to connect to the VM interface
+
+#### UNRESOLVED Summary discussions
+
+- Figure out details of exec probe performance problem.
+
+- Figure out if network probe performance is still a problem.
+
+- Figure out if there are any Windows-specific concerns (e.g, that our
+  proposal may be more difficult on Windows). Alternatively, it may
+  turn out that removing the constraint that kubelet needs to be able
+  to talk to all pods would simplify things on Windows?
+
+[kubernetes #99425]: https://github.com/kubernetes/kubernetes/issues/99425
+[AdminNetworkPolicy]: https://github.com/kubernetes/enhancements/blob/master/keps/sig-network/2091-admin-network-policy/README.md
+[future successor to NetworkPolicy]: https://github.com/kubernetes-sigs/network-policy-api/issues/136
+[multi-tenant scenarios]: https://github.com/kubernetes/kubernetes/issues/31824
+[Multi-network feature]: https://github.com/kubernetes/enhancements/issues/3698
+[kubernetes #101324]: https://github.com/kubernetes/kubernetes/issues/101324
+[the NetworkPolicy has not yet been fully programmed]: https://kubernetes.io/docs/concepts/services-networking/network-policies/#pod-lifecycle
+[readiness gates]: https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-readiness-gate
+[Exec probes are much, much slower than network probes]: https://medium.com/netcracker/exec-probes-a-story-about-experiment-and-relevant-problems-12de616c0a76
+[kubernetes #89898]: https://github.com/kubernetes/kubernetes/issues/89898
+[kubernetes #115143]: https://github.com/kubernetes/kubernetes/pull/115143
+[kubernetes #102613]: https://github.com/kubernetes/kubernetes/issues/102613
+[kubevirt]: https://github.com/kubevirt/kubevirt/blob/main/docs/devel/networking.md#vmi-networking
+
+## Motivation
+
+### Goals
```

Comment:
Lifecycle hooks should be included as explicitly in-scope.

Reply:
yep, good point, thanks Tim! will update to include this